packages/cli/src/commands/register.ts

```diff
@@ -39,22 +41,29 @@ export async function register({ cliSettings, options, packageRef, fromPublish }
   const [optimismRegistryConfig, mainnetRegistryConfig] = cliSettings.registries;
   const [optimismRegistryProvider, mainnetRegistryProvider] = await resolveRegistryProviders(cliSettings);
 
-  const isRegisteredOnMainnet = await isPackageRegistered(
-    [mainnetRegistryProvider],
-    packageRef,
-    mainnetRegistryConfig.address
+  // if any of the packages are registered, throw an error
+  const isRegistered = await Promise.all(
+    packageRef.map(async (pkg: PackageSpecification) => {
+      const isRegisteredOnMainnet = await isPackageRegistered(
+        [mainnetRegistryProvider],
+        pkg.name,
+        mainnetRegistryConfig.address
+      );
+
+      const isRegisteredOnOptimism = await isPackageRegistered(
+        [optimismRegistryProvider],
+        pkg.name,
+        optimismRegistryConfig.address
+      );
+
+      if (isRegisteredOnMainnet && isRegisteredOnOptimism) {
+        throw new Error(`The package "${pkg.name}" is already registered.`);
+      }
+
+      return [isRegisteredOnMainnet, isRegisteredOnOptimism];
+    })
```

Comment:
The `isPackageRegistered` function could be improved to be able to take one `contractAddress` per provider, and this block be removed almost completely. (it would also parallelize both calls and have a small speed improvement)